line-sdk/src/main/java/com/linecorp/linesdk/internal/nwclient/TalkApiClient.java

```diff
@@ -240,52 +245,134 @@ protected TalkApiClient(
         String postData;
         try {
             postData = MessageSendRequest.createOttType(ott, messages).toJsonString();
         } catch (JSONException e) {
-            return LineApiResponse.createAsError(
-                    LineApiResponseCode.INTERNAL_ERROR,
-                    new LineApiError(e));
+            return createInternalErrorResponse(e);
         }
 
         return httpClient.postWithJson(
                 buildUri(apiBaseUrl, BASE_PATH_MESSAGE_API, PATH_OTT_SHARE),
                 buildRequestHeaders(accessToken),
                 postData,
-                new MultiSendResponseParser());
+                MULTI_SEND_RESPONSE_PARSER);
+    }
+
+    @NonNull
+    public LineApiResponse<Boolean> updateAgreementStatus(
+            @NonNull InternalAccessToken accessToken,
+            @NonNull Boolean agreed) {
+        final Uri uri = buildUri(apiBaseUrl, BASE_PATH_OPENCHAT_API, "terms/agreement");
+        final String postData = String.format("{ \"agreed\": %s }", (agreed) ? "true" : "false");
+
+        return httpClient.putWithJson(
+                uri,
+                buildRequestHeaders(accessToken),
+                postData,
+                null);
+    }
+
+    @NonNull
+    public LineApiResponse<OpenChatRoomInfo> createOpenChatRoom(
+            @NonNull InternalAccessToken accessToken,
+            @NonNull OpenChatParameters openChatParameters) {
+        final Uri uri = buildUri(apiBaseUrl, BASE_PATH_OPENCHAT_API, "square");
+
+        return httpClient.postWithJson(
+                uri,
+                buildRequestHeaders(accessToken),
+                openChatParameters.toJsonString(),
+                OPEN_CHAT_ROOM_INFO_PARSER);
+    }
+
+    @NonNull
+    public LineApiResponse<OpenChatRoomStatus> getOpenChatRoomStatus(
+            @NonNull InternalAccessToken accessToken,
+            @NonNull String roomId) {
+        final Uri uri = buildUri(apiBaseUrl, BASE_PATH_OPENCHAT_API, "square", roomId, "status");
+        final Map<String, String> queryParameters = new HashMap<>();
+        queryParameters.put("squareMid", roomId);
+
+        return httpClient.get(
+                uri,
+                buildRequestHeaders(accessToken),
+                queryParameters,
+                OPEN_CHAT_ROOM_STATUS_PARSER);
+    }
+
+    @NonNull
+    public LineApiResponse<MembershipStatus> getMembershipStatus(
+            @NonNull InternalAccessToken accessToken,
+            @NonNull String roomId) {
+        final Uri uri = buildUri(apiBaseUrl, BASE_PATH_OPENCHAT_API, "square", roomId, "membership");
+
+        return httpClient.get(
+                uri,
+                buildRequestHeaders(accessToken),
+                Collections.emptyMap(),
+                OPEN_CHAT_MEMBERSHIP_PARSER);
+    }
+
+    private LineApiResponse createInternalErrorResponse(Exception exception) {
+        return LineApiResponse.createAsError(LineApiResponseCode.INTERNAL_ERROR, new LineApiError(exception));
+    }
```

Comment:
We can use a Generic Method to avoid unchecked assignment warning.

<img width="884" alt="螢幕快照 2020-01-14 上午11 25 10" src="https://user-images.githubusercontent.com/26949910/72311773-bbc39800-36c0-11ea-9a6b-a7482cc724eb.png">

```kotlin
private <T> LineApiResponse<T> createInternalErrorResponse(Exception exception) {
    return LineApiResponse.createAsError(LineApiResponseCode.INTERNAL_ERROR, new LineApiError(exception));
}
```